### ISSUE TITLE
Exception in Windows 8 when using Windows Forms, anti dump & Prefer32Bit set to False

### ISSUE THREAD

**Issue by BahKoo** [yck1509/ConfuserEx#458]
This can be reproduced by creating a new Windows Forms Project and enabling the "Prefer 32-bit" setting. Confuse the project with the following setting:

`<protection id="anti dump" />`

The application runs on Windows 7 but throws an exception on Windows 8.

**Comment by yck1509** [comment]
Hi,
I just download a VM of Windows 8 and I could not reproduce the issue. Do you have a debug stack trace of the exception?

**Comment by BahKoo** [comment]
Sorry. I assumed it would be easy to reproduce since I could reproduce it on multiple Window 8 machines. I attached to project I created, two entries in the windows event log and a debug session from WinDbg. Please let me know if you need more information. In the meantime, if you provide the link for windows 8 VM you used I can test it on that VM.

[ConfuserTest.zip](https://github.com/yck1509/ConfuserEx/files/180919/ConfuserTest.zip)
[exceptions.txt](https://github.com/yck1509/ConfuserEx/files/180920/exceptions.txt)
[debug.txt](https://github.com/yck1509/ConfuserEx/files/180921/debug.txt)

**Comment by yck1509** [comment]
I used the Win8 VM at https://dev.windows.com/en-us/microsoft-edge/
Could you try changing the test app to Console Application, and enabling the generate debug symbol option, then send me the exception stack trace it print?

**Comment by BahKoo** [comment]
Here is the exception stack when running the app as a Console App:

```
Unhandled Exception: System.TypeInitializationException: The type initializer for '<Module>' threw an exception. ---> System.AccessViolationException: Attempted to read or write protected memory. This is often an indication that other memory is corrupt.
   at ?????????????????????????????????????????() in e:\Source\Public\Confuser2\Confuser.Runtime\AntiDump.cs:line 75
   at .cctor()
   --- End of inner exception stack trace ---
```

**Comment by BahKoo** [comment]
I could not reproduce the error when testing with the the Win8 VM at https://dev.windows.com/en-us/microsoft-edge/
However, I was able to reproduce the problem on two separate Win8 machines. If you would like me to gather more info about those machines, let me know.

**Comment by BahKoo** [comment]
It looks like this issue may have been fixed with commit [a4a8856](a4a885680f6fdf2325f3d50a078df71c8ba7343b). I was testing with the 0.6.0 release and not the latest build from master. I can test and let you know.

**Comment by BahKoo** [comment]
Confirmed. The issue is already fixed in master. Next time I'll test master before reporting an issue.
